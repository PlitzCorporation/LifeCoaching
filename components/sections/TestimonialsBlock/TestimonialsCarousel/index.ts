import TestimonialsCarousel from './TestimonialsCarousel';

export default TestimonialsCarousel;
