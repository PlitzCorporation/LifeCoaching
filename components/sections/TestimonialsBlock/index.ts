import TestimonialsBlock from './TestimonialsBlock';

export default TestimonialsBlock;
