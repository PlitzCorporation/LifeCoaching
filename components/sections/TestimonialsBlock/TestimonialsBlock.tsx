import React from 'react';

import { getAllTestimonials } from '@/helpers/apiCalls/pullTestimonialsData';

import { WhiteThickLine } from '@/components/ui/decorations';
import TestimonialsCarousel from './TestimonialsCarousel';

import { TestimonialProps } from '@/types/testimonials';

const TestimonialsBlock = async () => {
	const testimonials: TestimonialProps[] = await getAllTestimonials();
	const testimonialsData = testimonials
		.sort(
			(a, b) =>
				// new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime(),
				a.order - b.order,
		)
		.slice(0, 5);

	return (
		<section className="bg-gradient-to-b from-plitz-primary to-[#4B1322] py-16 lg:py-32">
			<div className="max-w-screen-xl plitz-container text-center drop-shadow-plitz-darker text-white text-balance">
				<div className="w-max mx-auto mb-8">
					<WhiteThickLine />
				</div>
				<div className="text-white font-bold mt-7 lg:mt-10 mb-4">
					Testimonials
				</div>
				<h3 className="mb-11 lg:mb-20">We are happy to help you!</h3>
				<TestimonialsCarousel latestTestimonials={testimonialsData} />
			</div>
		</section>
	);
};

export default TestimonialsBlock;
