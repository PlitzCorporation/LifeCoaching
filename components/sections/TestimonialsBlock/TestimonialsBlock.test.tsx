import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import TestimonialsBlock from './TestimonialsBlock';

jest.mock('react-slick', () => ({
	__esModule: true,
	default: ({ children }: { children: React.ReactNode }) => (
		<div className="slick-slider slick-initialized">
			<div className="slick-arrow slick-prev" style={{ display: 'block' }}>
				&#x2190;
			</div>
			<div className="slick-list">{children}</div>
			<div className="slick-arrow slick-prev" style={{ display: 'block' }}>
				&#x2192;
			</div>
			<div className="slick-dots">
				<ul className="flex gap-2 lg:gap-4 justify-center w-full flex-wrap items-center h-max">
					<li className="">
						<div className="h-3 w-3 bg-plitz-secondary rounded-full"></div>
					</li>
					<li className="slick-active">
						<div className="h-3 w-3 bg-plitz-secondary rounded-full"></div>
					</li>
					<li className="">
						<div className="h-3 w-3 bg-plitz-secondary rounded-full"></div>
					</li>
				</ul>
			</div>
		</div>
	),
}));

// Mock the getAllTestimonials function
jest.mock('@/helpers/apiCalls/pullTestimonialsData', () => ({
	getAllTestimonials: jest.fn().mockResolvedValue([
		{
			_id: '1',
			isLive: true,
			order: 1,
			author: 'Test User 1',
			details: 'Great service!',
			createdAt: '2023-09-01T00:00:00Z',
		},
		{
			_id: '2',
			isLive: true,
			order: 2,
			author: 'Test User 2',
			details: 'Amazing experience!',
			createdAt: '2023-09-03T00:00:00Z',
		},
		{
			_id: '3',
			isLive: true,
			order: 3,
			author: 'Test User 3',
			details: 'Highly recommend!',
			createdAt: '2023-08-30T00:00:00Z',
		},
	]),
}));

describe('TestimonialsBlock', () => {
	it('should render the TestimonialsBlock with sorted and sliced testimonials', async () => {
		const ComponentResolved = await TestimonialsBlock();

		const MockComponent = () => ComponentResolved;
		const { container } = render(<MockComponent />);

		// Check if the testimonials carousel has been rendered
		const heading = screen.getByText('Great service!');
		expect(heading).toBeInTheDocument();

		// Check if the latest testimonial's details appears in the carousel
		const testimonialMessage = screen.getByText('Amazing experience!');
		expect(testimonialMessage).toBeInTheDocument();

		// Assert
		expect(container).toMatchSnapshot();
	});
});
