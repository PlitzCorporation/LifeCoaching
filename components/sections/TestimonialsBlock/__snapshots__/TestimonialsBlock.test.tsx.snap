// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestimonialsBlock should render the TestimonialsBlock with sorted and sliced testimonials 1`] = `
<div>
  <section
    class="bg-gradient-to-b from-plitz-primary to-[#4B1322] py-16 lg:py-32"
  >
    <div
      class="max-w-screen-xl plitz-container text-center drop-shadow-plitz-darker text-white text-balance"
    >
      <div
        class="w-max mx-auto mb-8"
      >
        <div
          class="block w-32 h-1 bg-gradient-to-r from-white/5 to-white/100"
        />
      </div>
      <div
        class="text-white font-bold mt-7 lg:mt-10 mb-4"
      >
        Testimonials
      </div>
      <h3
        class="mb-11 lg:mb-20"
      >
        We are happy to help you!
      </h3>
      <div
        class="mb-20 plitz-container"
      >
        <div
          class="slick-slider slick-initialized"
        >
          <div
            class="slick-arrow slick-prev"
            style="display: block;"
          >
            ←
          </div>
          <div
            class="slick-list"
          >
            <div
              class="flex flex-col px-10 text-center w-full"
            >
              <blockquote
                class="text-xs lg:text-sm text-white leading-6 lg:leading-7 mb-7 lg:mb-10"
              >
                Great service!
              </blockquote>
              <small
                class="text-plitz-accent"
              >
                Test User 1
              </small>
            </div>
            <div
              class="flex flex-col px-10 text-center w-full"
            >
              <blockquote
                class="text-xs lg:text-sm text-white leading-6 lg:leading-7 mb-7 lg:mb-10"
              >
                Amazing experience!
              </blockquote>
              <small
                class="text-plitz-accent"
              >
                Test User 2
              </small>
            </div>
            <div
              class="flex flex-col px-10 text-center w-full"
            >
              <blockquote
                class="text-xs lg:text-sm text-white leading-6 lg:leading-7 mb-7 lg:mb-10"
              >
                Highly recommend!
              </blockquote>
              <small
                class="text-plitz-accent"
              >
                Test User 3
              </small>
            </div>
          </div>
          <div
            class="slick-arrow slick-prev"
            style="display: block;"
          >
            →
          </div>
          <div
            class="slick-dots"
          >
            <ul
              class="flex gap-2 lg:gap-4 justify-center w-full flex-wrap items-center h-max"
            >
              <li
                class=""
              >
                <div
                  class="h-3 w-3 bg-plitz-secondary rounded-full"
                />
              </li>
              <li
                class="slick-active"
              >
                <div
                  class="h-3 w-3 bg-plitz-secondary rounded-full"
                />
              </li>
              <li
                class=""
              >
                <div
                  class="h-3 w-3 bg-plitz-secondary rounded-full"
                />
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
